src/resilience/Config.hpp

```diff
@@ -174,30 +172,29 @@ namespace KokkosResilience
       template< typename T >
       const T &as() const
       {
-        if ( m_variant.which() != 2 )
+        auto *val = std::get_if< Value >( &m_variant );
+        if ( !val )
           throw ConfigValueError();
 
-        const auto &val = boost::get< Value >( m_variant );
-        return val.as< T >();
+        return val->as< T >();
       }
 
       bool is_value() const noexcept
       {
-        return m_variant.which() == 2;
+        return std::holds_alternative< Value >( m_variant );
       }
 
       bool is_object() const noexcept
       {
-        return m_variant.which() == 0;
+        return std::holds_alternative< map_type >( m_variant );
       }
 
     private:
 
       using map_type = std::unordered_map< std::string, Entry >;
       using array_type = std::vector< Entry >;
 
-      boost::variant< boost::recursive_wrapper< map_type >,
```

Comment:
I'm getting an error here in the map_type, which uses a key/val tuple that includes Entry. 
```
stl_pair.h: In instantiation of 'struct std::pair<const std::__cxx11::basic_string<char>, KokkosResilience::Config::Entry>'
```
Entry isn't yet fully defined.
```
stl_pair.h:218:11: error: 'std::pair<_T1, _T2>::second' has incomplete type
```
Probably solved by making this a unique pointer to the variant?

Reply:
Fixed by making it a regular map, unfortunately the value has to be copied so unique_ptr doesn't work here